components/count-badge/test/count-badge-icon.test.js

```diff
@@ -1,9 +1,135 @@
 import '../count-badge-icon.js';
-import { runConstructor } from '@brightspace-ui/testing';
+import { expect, fixture, html, runConstructor } from '@brightspace-ui/testing';
 
 describe('d2l-count-badge-icon', () => {
-	it('should construct', () => {
-		runConstructor('d2l-count-badge-icon');
+
+	describe('constructor', () => {
+		it('should construct', () => {
+			runConstructor('d2l-count-badge-icon');
+		});
+	});
+
+	describe('defaults', () => {
+
+		it('should default announceChanges to false', async() => {
+			const el = await fixture(html`<d2l-count-badge-icon icon="tier3:gear" number="1" text="1 notification"></d2l-count-badge-icon>`);
+			expect(el.announceChanges).to.be.false;
+		});
+
+	});
+
+	describe('number formatting', () => {
+
+		it('should display number as-is when within maxDigits', async() => {
+			const el = await fixture(html`<d2l-count-badge-icon icon="tier3:gear" number="42" text="42 items"></d2l-count-badge-icon>`);
+			const numberString = el.getNumberString();
+			expect(numberString).to.equal('42');
+		});
+
+		it('should truncate and add plus when exceeding maxDigits for notification type', async() => {
+			const el = await fixture(html`<d2l-count-badge-icon icon="tier3:gear" type="notification" number="150" text="150 notifications"></d2l-count-badge-icon>`);
+			await el.updateComplete;
+			const numberString = el.getNumberString();
+			expect(numberString).to.equal('99+');
+		});
+
+		it('should truncate count type at 5 digits', async() => {
+			const el = await fixture(html`<d2l-count-badge-icon icon="tier3:gear" type="count" number="123456" text="items"></d2l-count-badge-icon>`);
+			await el.updateComplete;
+			const numberString = el.getNumberString();
+			expect(numberString).to.equal('99,999+');
+		});
+
+		it('should use custom maxDigits when provided', async() => {
+			const el = await fixture(html`<d2l-count-badge-icon icon="tier3:gear" max-digits="3" number="5000" text="5000 items"></d2l-count-badge-icon>`);
+			const numberString = el.getNumberString();
+			expect(numberString).to.equal('999+');
+		});
+
+		it('should clamp maxDigits to 5 when set higher', async() => {
+			const el = await fixture(html`<d2l-count-badge-icon icon="tier3:gear" max-digits="10" number="1" text="1 item"></d2l-count-badge-icon>`);
+			await el.updateComplete;
+			expect(el.maxDigits).to.equal(5);
+		});
+
+		it('should clamp maxDigits to 5 when updated to higher value', async() => {
```

Comment:
TBH, I'm not sure this adds much value beyond the previous one. Beyond the previous, it seems like it's just making sure that Lit's property assignment is working.